beetsplug/lyrics.py

```diff
@@ -659,22 +663,69 @@ def commands(self):
             help=u'print lyrics to console',
         )
         cmd.parser.add_option(
+            u'-r', u'--write-rst', dest='writerst',
+            action='store', default='.',
+            help=u'write lyrics to given directory as RST files',
+        )
+        cmd.parser.add_option(
             u'-f', u'--force', dest='force_refetch',
             action='store_true', default=False,
             help=u'always re-download lyrics',
         )
+        cmd.parser.add_option(
+            u'-s', u'--skip', dest='skip_fetched',
+            action='store_true', default=False,
+            help=u'skip already fetched lyrics',
+        )
 
         def func(lib, opts, args):
             # The "write to files" option corresponds to the
             # import_write config value.
             write = ui.should_write()
+            artist = u'Unknown artist'
+            album = False
+            output = sys.stdout
             for item in lib.items(ui.decargs(args)):
-                self.fetch_item_lyrics(
-                    lib, item, write,
-                    opts.force_refetch or self.config['force'],
-                )
-                if opts.printlyr and item.lyrics:
-                    ui.print_(item.lyrics)
+                if not opts.skip_fetched and not self.config['skip']:
+                    self.fetch_item_lyrics(
+                        lib, item, write,
+                        opts.force_refetch or self.config['force'],
+                    )
+                if item.lyrics:
+                    if opts.printlyr:
+                        ui.print_(item.lyrics)
+                    if opts.writerst:
```

Comment:
i can see moving it to a different section, but the problem is that we'd need to iterate over all the items another time - we wouldn't write the lyrics as soon as we download them, whereas we do for the `--print` function.

we need a complete loop access because of the way we open that rst file: one per artist means we need to close the file outside of the loop when we're done.

unless i missed something obvious here of course. been staring at that loop too long already. ;)